omnibus/files/private-chef-ctl-commands/rotate_credentials.rb

```diff
@@ -122,6 +137,77 @@
   exit(0)
 end
 
+add_command_under_category "require-credential-rotation", "credential-rotation", "Disable the Chef Server and require credential rotation", 2 do
+  @agree_to_disable = false
+  @ui = HighLine.new
+
+  OptionParser.new do |opts|
+    opts.banner = "Usage: chef-server-ctl require-credential-rotation [--yes]"
+
+    opts.on("-h", "--help", "Show this message") do
+      puts opts
+      exit
+    end
+
+    opts.on("-y", "--yes", "Agree to disable the Chef Server and require credential rotation") do
+      @agree_to_disable = true
+    end
+  end.parse!(ARGV)
+
+  # Agree to disable and require rotation
+  msg = "Are you sure you want to disable the Chef Server and require credential rotation?" \
+          "\n Type 'yes' to confirm: "
+  exit(1) unless @agree_to_disable || @ui.ask("<%= color(%Q(#{msg}), :yellow) %>") =~ /yes/i
+
+  # Shut down Chef Server
+  run_sv_command("stop")
+
+  # Disable services
+  get_all_services.each do |service|
+    File.unlink("#{service_path}/#{service}") if service_enabled?(service)
+  end
+
+  # Create global pre hook that'll prevent ctl commands from running until the
+  # credentials have been rotated
+  File.open(credential_prehook_file_path, "a+") do |file|
+    file.write <<-EOF.gsub(/^\s{6}/, "")
+      add_global_pre_hook "require_credential_rotation" do
+        # Allow running "chef-server-ctl rotate-shared-secrets"
+        return true if ARGV.include?("opscode") && ARGV.include?("rotate-shared-secrets")
+
+        raise("You must rotate the Chef Server credentials to enable the Chef Server. "\
```

Comment:
I'm not sure I understand the intent of this command. It seems like it just stops the chef server and locks it. How is this desirable from a user's standpoint?